tsconfig.json

```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    "target": "ES2020" /* ES2020 should be safe to use in browsers */,
+    "module": "NodeNext" /* Required for ESM mode */,
+    "lib": ["DOM", "ESNext"] /* Specify library files to be included in the compilation. */,
+
+    /* Strict Type-Checking Options */
+    "strict": true /* Enable all strict type-checking options. */,
```

Comment:
There are more options that we can enable to increase strictness that we should probably enable:

- https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes
- https://www.typescriptlang.org/tsconfig#noFallthroughCasesInSwitch
- https://www.typescriptlang.org/tsconfig#noImplicitOverride
- https://www.typescriptlang.org/tsconfig#noImplicitReturns
- https://www.typescriptlang.org/tsconfig#noPropertyAccessFromIndexSignature
- https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess
- https://www.typescriptlang.org/tsconfig#noUnusedLocals
- https://www.typescriptlang.org/tsconfig#noUnusedParameters
- https://www.typescriptlang.org/tsconfig#allowUnusedLabels
- https://www.typescriptlang.org/tsconfig#allowUnreachableCode

Reply:
I've enabled:
-    "noUncheckedIndexedAccess": true,
-    "exactOptionalPropertyTypes": true,
-    "noPropertyAccessFromIndexSignature": true,
-    "noImplicitOverride": true,

`noUnusedLocals` is checked during explicit lint, and the others should already be checked by eslint.

Reasons for doing them in eslint are:
- we can be sure that eslint is correctly configured everywhere, whereas for tsconfig.json it's less likely
- ts-disable can only disable complete lines, not individual rules